application/src/main/scala/com/azavea/franklin/crawler/CatalogStacImport.scala

```diff
@@ -239,44 +239,42 @@ class CatalogStacImport(val catalogRoot: String) {
       StacIO.collectionCatalogFallback(absHref, collectionAttempt, catalogAttempt) flatMap {
         case (links, collectionO) =>
-          val filtered =
-            links.filter(_.rel == StacLinkType.Item)
+          val filtered = links.filter(_.rel == StacLinkType.Item)
           logger
             .debug(s"Links to read for $absHref: ${filtered map { _.href }}") <* (filtered traverse {
             link =>
               val itemPath = makeAbsPath(absHref, link.href)
-              readItem(itemPath, true, collectionO) flatMap {
-                item =>
-                  logger
-                    .debug(s"Item links after read for ${item.id}: ${item.links map { _.href }}") *>
-                    createCollectionForGeoJsonAsset(
-                      item,
-                      itemPath
-                    ) flatMap { newAssets =>
-                    val assets   = newAssets map { _._1 }
-                    val wrappers = newAssets map { _._2 }
-                    logger.debug(s"Item ${item.id} has ${newAssets.length} additional assets") *>
-                      logger.debug(s"Item links before write: ${item.links map { _.href }}") *>
-                      (StacItemDao
-                        .insertStacItem(
-                          item
-                            .copy(
-                              assets =
-                                item.assets ++ assets.foldK
-                            )
-                        ) *>
-                        (wrappers traverse { collectionWrapper =>
-                          insertCollection(collectionWrapper)
-                        }))
-                        .transact(xa)
-                  }
+              readItem(itemPath, true, collectionO) flatMap { item =>
+                logger
+                  .debug(s"Item links after read for ${item.id}: ${item.links map { _.href }}") *>
+                  createCollectionForGeoJsonAsset(
+                    item,
+                    itemPath
+                  ) flatMap { newAssets =>
+                  val assets   = newAssets map { _._1 }
+                  val wrappers = newAssets map { _._2 }
+                  logger.debug(s"Item ${item.id} has ${newAssets.length} additional assets") *>
+                    logger.debug(s"Item links before write: ${item.links map { _.href }}") *>
+                    (StacItemDao
+                      .insertStacItem(
+                        item
+                          .copy(
+                            assets =
+                              item.assets ++ assets.foldK
+                          )
+                      ) *>
+                      (wrappers traverse { collectionWrapper =>
+                        insertCollection(collectionWrapper)
+                      }))
+                      .transact(xa)
+                }
               }
           })
       }
-    }).void
+    }.flatten
```

Comment:
The actual return type here is `IO[IO[Unit]]`. `void`ing it would cause the internal IO loss.

Reply:
Yeah from the diff I saw what it was. A sad day for past me, that guy's got some real iffy ideas